### PR TITLE
satellite/satellitedb: Add piece count column to nodes table

### PR DESCRIPTION
**Pull request by bryanchriswhite**
What: Add piece count column to nodes table in satellitedb for use in garbage collection. 

Why: Part 1 of [V3-2157](https://storjlabs.atlassian.net/browse/V3-2157)

Please describe the tests: N/A
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/overlaycache.go [hunk 1]
```diff
@@ -636,6 +636,7 @@ func (cache *overlaycache) UpdateAddress(ctx context.Context, info *pb.Node, def
 			dbx.Node_TotalAuditCount(0),
 			dbx.Node_UptimeSuccessCount(0),
 			dbx.Node_TotalUptimeCount(0),
+			dbx.Node_PieceCount(0),
```
Comment (mobyvb) [inline satellite/satellitedb/overlaycache.go]: make sure to add the field to `overlaycache.Get` as well

Comment (bryanchriswhite) [reply satellite/satellitedb/overlaycache.go]: :+1:

File: satellite/satellitedb/migrate.go [hunk 2]
```diff
@@ -1121,6 +1121,13 @@ func (db *DB) PostgresMigration() *migrate.Migration {
 					`DROP TABLE certRecords CASCADE`,
 				},
 			},
+			{
+				Description: "Add piece_count column to nodes table",
+				Version:     53,
+				Action: migrate.SQL{
+					`ALTER TABLE nodes ADD piece_count INTEGER NOT NULL DEFAULT 0;`,
```
Comment (ifraixedes) [inline satellite/satellitedb/migrate.go]: There isn't a default specified in the DBX model, so we have to specify it there or remove the `DEFAULT` constraint in a new SQL migration after this one.

Comment (bryanchriswhite) [reply satellite/satellitedb/migrate.go]: I looked through the dbx documentation and I don't see anything about setting defaults. Are you sure that's a thing?

Comment (bryanchriswhite) [reply satellite/satellitedb/migrate.go]: Nevermind, @mobyvb pointed out `autoinsert`.

File: satellite/satellitedb/testdata/postgres.v53.sql [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    id bytea NOT NULL,
+    address text NOT NULL,
+    last_net text NOT NULL,
+    protocol integer NOT NULL,
+    type integer NOT NULL,
+    email text NOT NULL,
+    wallet text NOT NULL,
+    free_bandwidth bigint NOT NULL,
+    free_disk bigint NOT NULL,
+    piece_count bigint NOT NULL,
```
Comment (ifraixedes) [inline satellite/satellitedb/testdata/postgres.v53.sql]: I don't know if this file should match the current default database schema if that's the case then it this column need to have the default

Comment (bryanchriswhite) [reply satellite/satellitedb/testdata/postgres.v53.sql]: I think dbx is responsible for setting the default with the `autoinsert`  attribute.

Comment (ifraixedes) [reply satellite/satellitedb/testdata/postgres.v53.sql]: Yess, that's right, this is more a question if these test schemas should match 100% the implementation DB schema.

**Pull request merged**